crates/solidity/outputs/npm/tests/src/tests/cst-cursor.ts

```diff
@@ -106,3 +106,30 @@ test("use cursor", () => {
   expectToken(cursor.node(), TokenKind.Semicolon, ";");
   expect(cursor.goToNext()).toBe(false);
 });
+
+test("cursor navigation", () => {
+  const data = "contract Foo {} contract Bar {} contract Baz {}";
+
+  const language = new Language("0.8.0");
+  const parseTree = language.parse(RuleKind.SourceUnit, data);
+
+  let contractNames = [];
+  let cursor = parseTree.createTreeCursor();
+
+  while (cursor.goToNextRuleWithKinds([RuleKind.ContractDefinition])) {
+    // You have to make sure you return the cursor to original position
+    cursor.goToFirstChild();
+    cursor.goToNextTokenWithKinds([TokenKind.Identifier]);
+
+    // The currently pointed-to node is the name of the contract
+    let tokenNode = cursor.node();
+    if (tokenNode.kind !== TokenKind.Identifier) {
```

Comment:
nit: I suggest using `expectToken()` like above.

Reply:
`expectToken` also validates the text value, whereas we collect them and assert at the end. Do you want me to modify `expectToken` to optionally handle the `text` parameter so that we can optionally assert that using this function?

Reply:
Using bare `expect` does not inform the TS type-checker the way usual if/else + diverging statement does. With
```ts
expect(tokenNode.kind).toEqual(TokenKind.Identifier);
```
it can't infer that the `tokenNode.kind` has to be `TokenKind.Identifier`, and so can't reason that `tokenNode` must be a `cst.TokenNode` (and therefore always has a `text` property).

Reply:
Sounds good. Or if you agree, maybe split into `expectToken(node, kind)` and `expectTokenText(node, kind, text)` for clarity.

Reply:
I moved the snippet to the doc examples in https://github.com/NomicFoundation/slang/pull/716/commits/d05e7577a022a362bb580554e12c313720876073 and decided against using our custom helper `expectToken` in the public snippets to show how a vanilla approach works; so I didn't change these helper functions in the end.